src/pages/index.js

```diff
@@ -166,11 +205,76 @@ function getCardElement(data) {
 const cardsList = document.querySelector(".cards__list");
 
 // Loop through the initialCards array and append each card to the parent
+/*
 initialCards.forEach((cardData) => {
   renderCard(cardData, "append");
 });
+*/
 
 function renderCard(item, method = "prepend") {
   const cardElement = getCardElement(item);
   cardsList[method](cardElement);
 }
+
+
+// functions
+console.log(deleteModal)
+deleteModalDeleteBtn.addEventListener('click', () => {
+  //console.log(currentCardToDelete)
+  deleteModalSubmitButton.textContent = 'Deleting...'
+  api.deleteCard(selectedCardId).then((dd) => {
+    selectedCardEl.remove()
+    closeModal(deleteModal)
+  }).catch((err) => console.error(err)).finally(() => deleteModalSubmitButton.textContent = 'Delete')
+})
+
+deleteModalCancleBtn.addEventListener('click', () => {
+  closeModal(deleteModal)
+})
+
+// user information
+function displayUserInformation() {
+  api.getUserInformation().then((dd) => {
+    profileNameElement.textContent = dd.name
+    profileJobElement.textContent = dd.about
+    profileAvatartElement.setAttribute('src', dd.avatar)
+    console.log(dd)
+  }).catch((err) => console.error(err))
+}
+
+// initial cards
+function displayInitialCards() {
+  api.getInitialCards().then((dd) => {
+    console.log(dd)
+    dd.forEach((cardData) => {
+      renderCard(cardData, "append");
+    });
+    }).catch((err) => console.error(err))
+}
+
+function handleAddCardSubmit(e) {
+  e.preventDefault();
+  cardModalSubmitButton.textContent = 'Saving...'
+  const data = {
+    name: cardNameInput.value,
+    link: cardLinkInput.value,
+  }
+  api.addNewCard(data).then((dd) => {
+      //console.log(dd)
+      renderCard(dd)
+}).catch((err) => console.error(err)).finally(() => cardModalSubmitButton.textContent = 'Save');
+  e.target.reset();
+  //disable submit button
+  disableButton(cardModalSubmitButton);
+  closeModal(cardModal);
```

Comment:
You need to close modals only in `then` blocks after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well. **Please, correct it everywhere in the code**